core/src/allocation.jl

```diff
@@ -13,8 +13,8 @@ function allocation_graph_used_nodes!(p::Parameters, allocation_network_id::Int)
         node_type = graph[node_id].type
         if node_type in [:user, :basin]
             push!(used_nodes, node_id)
-
-        elseif length(inoutflow_ids(graph, node_id)) > 2
+        elseif count(x -> true, inoutflow_ids(graph, node_id)) > 2
```

Comment:
Why is that preferable? I've assumed that `length` simply consumes unsized iterators, just like your code using `count`.

Reply:
I see :)